lib/JavascriptModulesPlugin.js

```diff
@@ -124,9 +124,10 @@ class JavascriptModulesPlugin {
 								}),
 							filenameTemplate,
 							pathOptions: {
+								chunk,
+								filename: { ext: ".js" },
```

Comment:
This could solve the issue for future module type additions, but it's proactive and maybe 'reverted' for now

Reply:
I think it's weird that `filename` can be string or object. I would only allow string for `filename` and omit it here. It's not used anyway.